sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/StateHolder.java

```diff
@@ -95,11 +106,43 @@ static boolean getLoadStateFeatureFlag(String name) {
     static void setLoadState(String name, Boolean loaded) {
         LOAD_STATE.put(name, loaded);
     }
-    
+
     /**
      * @param LOAD_STATE the loadState to set
      */
     static void setLoadStateFeatureFlag(String name, Boolean loaded) {
         setLoadState(name + FEATURE_ENDPOINT, loaded);
     }
+
+    /**
+     * @return the nextForcedRefresh
+     */
+    public static Instant getNextForcedRefresh() {
+        return nextForcedRefresh;
+    }
+
+    /**
+     * Set after load or refresh is successful.
+     * @param nextForcedRefresh the nextForcedRefresh to set
+     */
+    public static void setNextForcedRefresh(Duration refreshPeriod) {
+        nextForcedRefresh = Instant.now().plusSeconds(refreshPeriod.getSeconds());
+    }
+
+    /**
+     * Sets a minimum value until the next refresh. If a refresh interval has passed or is smaller than the calculated
+     * backoff time, the refresh interval is set to the backoff time.
+     * @param refreshInterval period between refreshe checks.
+     * @param properties Provider properties for min and max backoff periods.
+     */
+    static void resetAll(Duration refreshInterval, AppConfigurationProviderProperties properties) {
+        nextForcedRefresh = CalculatedBackoffTime.calculateBefore(CLIENT_ENDPOINT, nextForcedRefresh, refreshInterval, properties);
+        for (Entry<String, State> entry : STATE.entrySet()) {
+            Instant newRefresh = CalculatedBackoffTime.calculateBefore(entry.getKey(), entry.getValue().getNextRefreshCheck(),
```

Comment:
If you store attempts in `STATE.entrySet`, you can pass the attempt number of each endpoint while calling `CalculatedBackoffTime.calculateBefore(..., ..., ...,  entry.getValue().getAttempts())`

Reply:
I have updated all of this to just pass in the state.